File: 3206-find-common-elements-between-two-arrays/find-common-elements-between-two-arrays.swift
class Solution {
    func findIntersectionValues(_ nums1: [Int], _ nums2: [Int]) -> [Int] {
        var result1 = 0
        var result2 = 0

        nums1.forEach {
            if nums2.contains($0) { result1 += 1 }
        }
    
        nums2.forEach {
            if nums1.contains($0) { result2 += 1 }
        }
    
        return [result1, result2]

    }
}

File: 3206-find-common-elements-between-two-arrays/find-common-elements-between-two-arrays.kt
class Solution {
    fun findIntersectionValues(nums1: IntArray, nums2: IntArray): IntArray {
        val arr1 = nums1.map {
            if (nums2.contains(it)) 1 else 0
        }.sumOf { it }

        val arr2 = nums2.map {
            if (nums1.contains(it)) 1 else 0
        }.sumOf { it }

        return intArrayOf(arr1, arr2)
    }
}